File: Subclass/FoodListNSTableView.m
#import "FoodListNSTableView.h"
#import "FoodDragSupportBase.h"

@implementation FoodListNSTableView

    -(void)awakeFromNib
    {
    //Register for drag operations
    [ self registerForDraggedTypes: [ NSArray arrayWithObject: FoodDataDragType ] ];

    }

    - (void)DeleteSelectedObjectInTableView:(id)sender
    {
    [ FoodListController remove: sender ];

    //This was just for testing other row heights can be removed
    //[ self setRowHeight: [ self rowHeight ] + 1 ];
    }



//- (NSTableRowView *)tableView:(NSTableView *)tableView rowViewForRow:(NSInteger)row
//{
//    NSLog(@"row =%ld",row);
//    JBDCustomRow *rowView = [[JBDCustomRow alloc]init];
//    return rowView;
//}

@end

File: Subclass/FoodListNSTableView.swift
//
//  FoodListNSTableView.swift
//  BeFit
//
//  Created by Andrej Slegl on 11/4/19.
//

import Foundation
import Cocoa

extension NSPasteboard.PasteboardType {
    static let foodDataDragType = Self("FoodDataDragType")
}

class FoodListNSTableView: NSTableView,
NSTableViewDelegate {
    @IBOutlet var FoodListController: AnyObject!
    
    override func awakeFromNib() {
        super.awakeFromNib()
        
        registerForDraggedTypes([.foodDataDragType])
    }
    
    @IBAction func DeleteSelectedObjectInTableView(_ sender: Any?) {
        FoodListController.remove(sender)
    }
    
    //- (NSTableRowView *)tableView:(NSTableView *)tableView rowViewForRow:(NSInteger)row
    //{
    //    NSLog(@"row =%ld",row);
    //    JBDCustomRow *rowView = [[JBDCustomRow alloc]init];
    //    return rowView;
    //}
}
